code/modules/reagents/reagent_containers/drinkingglass/glass_boxes.dm

```diff
@@ -13,69 +18,80 @@
-	glass_type = /obj/item/chems/drinks/glass2/square
+/obj/item/storage/box/glasses/square/WillContain()
+	return list(/obj/item/chems/drinks/glass2/square = storage_slots)
 
-/obj/item/storage/box/glasses/rocks
-	name = "box of rocks glasses"
-	glass_type = /obj/item/chems/drinks/glass2/rocks
+/obj/item/storage/box/glasses/rocks/WillContain()
+	return list(/obj/item/chems/drinks/glass2/rocks = storage_slots)
 
-/obj/item/storage/box/glasses/shake
-	name = "box of milkshake glasses"
-	glass_type = /obj/item/chems/drinks/glass2/shake
+/obj/item/storage/box/glasses/shake/WillContain()
+	return list(/obj/item/chems/drinks/glass2/shake = storage_slots)
 
-/obj/item/storage/box/glasses/cocktail
-	name = "box of cocktail glasses"
-	glass_type = /obj/item/chems/drinks/glass2/cocktail
+/obj/item/storage/box/glasses/cocktail/WillContain()
+	return list(/obj/item/chems/drinks/glass2/cocktail = storage_slots)
 
-/obj/item/storage/box/glasses/shot
-	name = "box of shot glasses"
-	glass_type = /obj/item/chems/drinks/glass2/shot
+/obj/item/storage/box/glasses/shot/WillContain()
+	return list(/obj/item/chems/drinks/glass2/shot = storage_slots)
 
-/obj/item/storage/box/glasses/pint
-	name = "box of pint glasses"
-	glass_type = /obj/item/chems/drinks/glass2/pint
+/obj/item/storage/box/glasses/pint/WillContain()
+	return list(/obj/item/chems/drinks/glass2/pint = storage_slots)
 
-/obj/item/storage/box/glasses/mug
-	name = "box of glass mugs"
-	glass_type = /obj/item/chems/drinks/glass2/mug
+/obj/item/storage/box/glasses/mug/WillContain()
+	return list(/obj/item/chems/drinks/glass2/mug = storage_slots)
 
-/obj/item/storage/box/glasses/wine
-	name = "box of wine glasses"
-	glass_type = /obj/item/chems/drinks/glass2/wine
+/obj/item/storage/box/glasses/wine/WillContain()
+	return list(/obj/item/chems/drinks/glass2/wine = storage_slots)
 
+////////////////////////////////////////////////////////////////////
+// Extras
+////////////////////////////////////////////////////////////////////
 /obj/item/storage/box/glass_extras
 	name = "box of cocktail garnishings"
-	var/extra_type = /obj/item/glass_extra
 	can_hold = list(/obj/item/glass_extra)
 	storage_slots = 14
 
-/obj/item/storage/box/glass_extras/Initialize()
-	for(var/i = 1 to 14)
-		new extra_type(src)
+/obj/item/storage/box/glass_extras/Initialize(ml, material_key)
 	. = ..()
-
-/obj/item/storage/box/glass_extras/straws
-	name = "box of straws"
-	extra_type = /obj/item/glass_extra/straw
-
-/obj/item/storage/box/glass_extras/sticks
-	name = "box of drink sticks"
-	extra_type = /obj/item/glass_extra/stick
+	setup_name()
+	if(length(contents))
+		make_exact_fit()
+
+/obj/item/storage/box/glass_extras/proc/setup_name()
+	//#TODO: This really should handle atom_creator stuff too..
+	var/list/cnt = WillContain()
+	if((islist(cnt) || ispath(cnt)) && length(cnt))
+		var/atom/movable/O = ispath(cnt) ? cnt : cnt[1]
+		SetName("box of [initial(O.name)]")  //#FIXME: Can't get plural from name.
```

Comment:
The note doesn't make that much sense, but what I meant was this for example: "box of pint glasses" instead of "box of pint glass". 
I just want to get the plural version of the kind of item in the box.

Reply:
If you add "s" after "pint glass" it just becomes "pint glasss" though?

Reply:
Don't think that would be enough though :
https://en.wikipedia.org/wiki/English_plurals#Regular_plurals
There's so many exceptions D:

Reply:
That couldn't handle a box of :
larva => larvae
pupa => pupae/pupas
or antennaes

Reply:
why not just "pint glass box"? that seems perfectly valid to me

Reply:
I mean, that would be simpler yeah. Except then every single other box is named "box of thing".

Reply:
if it's such an issue, rename those too? ;)

Reply:
Don't give the maint any ideas ;n;